Source/Core/VideoCommon/BPMemory.h

```diff
@@ -491,6 +491,81 @@ struct fmt::formatter<TevStageCombiner::ColorCombiner>
   template <typename FormatContext>
   auto format(const TevStageCombiner::ColorCombiner& cc, FormatContext& ctx)
   {
+    auto out = ctx.out();
+    if (cc.bias != TevBias::Compare)
+    {
+      // Generate an equation view, simplifying out addition of zero and multiplication by 1
+      // dest = (d (OP) ((1 - c)*a + c*b) + bias) * scale
+
+      // Note that this is simplified with regards to how lerping works; see PixelShaderGen's
+      // WriteTevRegular for more details.
+
+      static constexpr Common::EnumMap<const char*, TevColorArg::Zero> alt_names = {
+          "prev.rgb", "prev.aaa", "c0.rgb",  "c0.aaa",  "c1.rgb", "c1.aaa", "c2.rgb",    "c2.aaa",
+          "tex.rgb",  "tex.aaa",  "ras.rgb", "ras.aaa", "1",      ".5",     "konst.rgb", "0",
+      };
+
+      const bool has_d = cc.d != TevColorArg::Zero;
+      // If c is one, (1 - c) is zero, so (1-c)*a is zero
+      const bool has_a = cc.a != TevColorArg::Zero && cc.c != TevColorArg::One;
+      // If either b or c is zero, b*c is zero
+      const bool has_b = cc.b != TevColorArg::Zero && cc.c != TevColorArg::Zero;
```

Comment:
I think `has_b` is a bit clearer, in that the variable is indicating whether the `b` term shows up in the result, without saying anything about `c` (since it could appear in the `a` term instead).

`has_b_and_c` would be more accurate than `or` (since both b and c need to be nonzero), but there isn't a corresponding good name for `has_a`.

Reply:
Hmm, when you put it like that the current naming does make more sense. 

You could go more abstract with `have_left` and `have_right` for the two sides of the equation.

Reply:
I've ended up going with `has_bc` and `has_ac`, because on further thought that does still fit well enough in both cases.